src/lib/mina_base/snapp_basic.ml

```diff
@@ -119,7 +119,7 @@ module Set_or_keep = struct
   [%%ifdef consensus_mechanism]
 
   module Checked : sig
-    type 'a t
+    type 'a t = (Boolean.var, 'a) Flagged_option.t
```

Comment:
Revert this abstraction break?

Reply:
The reason for these abstraction leaks is that snarkyjs uses JS types built on top of plain `Field`s and `Bool`s. On the other hand, in OCaml we have these abstracted Checked types (which are Fields under the hood as well). There are situations where we need to convert the JS type -- say, a Bool plus a Field -- to the OCaml type, to use OCaml functions. And in these situations we need types like this one exposed.

Reply:
Can we use a `make_unsafe` function for creating these, and accessors for the members? It's fine if the SnarkyJS code needs to do this, but no other OCaml code should be able to

Reply:
Yes makes sense!